astropy/coordinates/sky_coordinate_parsers.py

```diff
@@ -588,7 +588,17 @@ def _get_representation_attrs(frame, units, kwargs):
     for frame_attr_name, repr_attr_class, unit in zip(frame_attr_names, repr_attr_classes, units):
         value = kwargs.pop(frame_attr_name, None)
         if value is not None:
-            valid_kwargs[frame_attr_name] = repr_attr_class(value, unit=unit)
+            try:
+                valid_kwargs[frame_attr_name] = repr_attr_class(value, unit=unit)
+            except u.UnitConversionError as err:
+                error_message = (
+                    f"Unit '{unit}' ({unit.physical_type}) could not be applied to '{frame_attr_name}'. "
+                    "This can occur when passing units for some coordinate components "
+                    "when other components are specified as Quantity objects. "
+                    "Either pass a list of units for all components (and unit-less coordinate data), "
+                    "or pass Quantities for all components"
```

Comment:
Period at the end of the final sentence?

Reply:
I just checked and the use of periods at the end of error messages in Astropy is rather inconsistent :sweat_smile: However, in this case there are two sentences, so I will add it

<sub>And leave no period at the end of this comment because I just want to see the world burn</sub>

Reply:
Haha! There was a discussion about adding or not adding period to errors/warnings at some point, but I cannot find it now. I wonder if @MSeifert04 remembers, if he is still around.